netket/vqs/mc/mc_state/state.py

```diff
@@ -83,6 +97,8 @@ class MCState(VariationalState):
     _apply_fun: Callable = None
     """The function used to evaluate the model"""
 
+    _chunk_size: Optional[int] = None
+
     def __init__(
```

Comment:
Atm, we only use `compute_chain_length` in the sample-number setter. I think we should use it in some way in the other relevant setters (`chunk_size`, and maybe `sampler`?) and also here in the constructor. Also, I don't like how we silently change the number of samples; we should give warnings somewhere (maybe in `compute_chain_length`?)

Reply:
Also, if I pass an incommensurate sample count to the constructor, it doesn't get adjusted when we `sample`, so `chunk_size`s get checked against a sample count that is not actually used!!

Reply:
OK, so if we want to ensure a number of samples that is commensurate with everything (modulo allowing big chunks), we should call something like my second suggestion above in all relevant setters and the constructor (maybe it should return `samples_per_rank` instead of `chain_length` at this point. Then, `sample` doesn't need to call `compute_chain_length` at all, but just divide `n_sample//n_chain`

Reply:
should be fine now, i think